### DISCUSSION TITLE
Add scriptable docker-entrypoint.d, like the nginx docker image provides

### DISCUSSION THREAD

**Discussion by devinrsmith** [deephaven/deephaven-core#1105] [Ideas]
The nginx docker image provides a place where shell scripts can be dropped to be always run during container startup, before the main process runs: https://github.com/nginxinc/docker-nginx/tree/master/entrypoint

This is useful to extend/customize/configure the nginx image for our own purposes. We use it to create our desired directory layout before nginx starts, https://github.com/deephaven/deephaven-core/blob/v0.3.0/web/client-ide/docker/Dockerfile#L19.

It may be useful to build a feature like this for our deephaven/grpc-api image. @nbauernfeind could potentially use it to bootstrap our application directory when necessary, and other users could extend it to easily install packages they want to use (python or java or other).

**Comment by rcaudy** [comment]
I like this. Can we similarly have a place to pass JVM args for the worker, and/or ports to open?

**Reply by devinrsmith** [reply]
Yes, we could use it for JVM args. I don't think it's at the right level of the stack to choose ports though, as that's a docker container orchestration concern.

**Comment by chipkent** [comment]
I think this would be a useful feature.  We should have something like this that supports (1) baking the "new stuff" into an image or (2) reading the "new stuff" from a locally mounted disk or volume.  If we are planning to add python this way, what we do should play nicely with pip and should not make the user hate us when they deal with it.
